WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -6,77 +6,136 @@ import {
   List,
 } from 'semantic-ui-react';
 import useLoadedData from '../../lib/hooks/useLoadedData';
+import useSaveAction from '../../lib/hooks/useSaveAction';
 import {
   roleListUrl,
+  currentUserUrl,
+  teamUrl,
 } from '../../lib/requests/routes.js.erb';
 import Errored from '../Requests/Errored';
 import Loading from '../Requests/Loading';
 import RoleForm from './RoleForm';
+import I18n from '../../lib/i18n';
+
+// let i18n-tasks know the key is used
+// i18n-tasks-use t('enums.user.role_status.trainee_delegate')
+// i18n-tasks-use t('enums.user.role_status.candidate_delegate')
+// i18n-tasks-use t('enums.user.role_status.delegate')
+// i18n-tasks-use t('enums.user.role_status.senior_delegate')
+// i18n-tasks-use t('enums.user.role_status.member')
+// i18n-tasks-use t('enums.user.role_status.senior_member')
+// i18n-tasks-use t('enums.user.role_status.leader')
+
+export default function RolesTab({ userId }) {
+  const loggedInUserFetch = useLoadedData(currentUserUrl);
+  const roleListFetch = useLoadedData(roleListUrl(userId));
+  const { data, sync } = roleListFetch;
+  const { save, saving } = useSaveAction();
 
-function RoleFormModal({
-  trigger, userId, isActiveRole, sync,
-}) {
   const [open, setOpen] = React.useState(false);
+  const [currentUserRoles, setCurrentUserRoles] = React.useState(null);
+  const [error, setError] = React.useState(null);
 
-  return (
-    <Modal
-      size="fullscreen"
-      onClose={() => {
-        setOpen(false);
-        sync();
-      }}
-      onOpen={() => setOpen(true)}
-      open={open}
-      trigger={trigger}
-    >
-      <Modal.Content>
-        <RoleForm
-          userId={userId}
-          isActiveRole={isActiveRole}
-        />
-      </Modal.Content>
-    </Modal>
+  const activeNonHiddenRoles = React.useMemo(
+    () => {
+      if (data) {
+        return data.activeRoles.filter((role) => role.group.is_hidden === false);
+      }
+      return [];
+    },
+    [data],
+  );
+  const isDelegate = React.useMemo(
```

Comment:
This doesn't need to be a useMemo.

If the return type of an expression is a primitive like a string, boolean, or number, then you don't need a useMemo.

Reply:
That's correct, using the `useMemo` will prevent it recalculating each re-render. But unless this array is _really_ big, or this component is being used _a lot_ of places at once, the performance is unlikely to be affected. At least, that's my understanding.

The other reason to memoize something is so that you can use it in a dependency array, but that's not applicable here because `true` is `true` and `false` is `false` (unlike `[]` which is not `[]`). Plus I don't think it's in a dependency array anyway.

Reply:
Okay got it.